lib/ansible/module_utils/vmware.py

```diff
@@ -1173,3 +1175,94 @@ def find_datastore_cluster_by_name(self, datastore_cluster_name):
             if dsc.name == datastore_cluster_name:
                 return dsc
         return None
+
+    def _extract(self, data, remainder):
+        """
+        This is used to break down dotted properties for extraction.
+
+        Args:
+            data (dict): result of _jsonify on a property
+            remainder: the remainder of the dotted property to select
+        """
+        if '.' not in remainder:
+            result = {}
+            result[remainder] = data[remainder]
+            return result
+        key, remainder = remainder.split('.', 1)
+        result = {}
+        result[key] = self._extract(data[key], remainder)
+        return result
+
+    def _jsonify(self, obj):
+        """ Convert something from pyVmomi into JSON. """
+        return json.loads(json.dumps(obj, cls=VmomiSupport.VmomiJSONEncoder))
+
+    def _strip_dynamic(self, data):
+        """
+        Entities contain 'dynamicProperty' and 'dynamicType'
+        fields which are usually empty.  Strip them from the
+        result if they are.
+
+        Args:
+            data (dict): results to strip
+        """
+        if isinstance(data, dict):
+            for key in data.keys():
+                if (key == 'dynamicType' and data[key] is None) or\
+                   (key == 'dynamicProperty' and len(data[key]) == 0):
+                    data.pop(key)
+                else:
+                    self._strip_dynamic(data[key])
+        elif isinstance(data, list):
+            for item in data:
+                self._strip_dynamic(item)
+
+    def managed_object_to_json(self, mobj, properties=None):
+        """
+        Convert a vSphere Managed Object into JSON.  This is a deep
+        transformation.  The list of properties is optional - if not
+        provided then all properties are deeply converted.
+
+        Requires upstream support from pyVmomi.
+
+        Args:
+            mobj (object): vim managed object
+            properties (list, optional): list of properties following
+                the property collector specification, for example:
+                ["config.hardware.memoryMB", "name", "overallStatus"]
+        """
+        if not hasattr(VmomiSupport, 'VmomiJSONEncoder'):
+            self.module.fail_json(msg='pyvmomi lacks JSON output support; see https://github.com/vmware/pyvmomi/pull/732')
+
+        result = {}
+        if properties:
+            for prop in properties:
+                try:
+                    if '.' in prop:
+                        key, remainder = prop.split('.', 1)
+                        tmp = {}
+                        tmp[key] = self._extract(self._jsonify(getattr(mobj, key)), remainder)
+                        _update(result, tmp)
+                    else:
+                        result[prop] = self._jsonify(getattr(mobj, prop))
+                except AttributeError:
```

Comment:
```suggestion
                except (AttributeError, KeyError) as e:
```

Reply:
Please ignore that, it is part of #47446 which I said needs to be merged first, then only the changes for this feature will be present.  This change depends on the vmware output schema that #47446 provides so it was branched off #47446.